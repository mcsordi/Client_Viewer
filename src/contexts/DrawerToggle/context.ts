import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
