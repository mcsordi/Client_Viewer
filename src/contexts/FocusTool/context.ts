import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
