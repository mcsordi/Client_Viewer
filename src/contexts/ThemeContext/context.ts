import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

