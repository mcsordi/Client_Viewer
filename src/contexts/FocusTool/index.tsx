import React, { useState } from 'react';
import { Children } from '../../shared/types/Children';
import { focusTool } from './context';

export const FocusTool: React.FC<Children> = ({ children }) => {
  const [focus, setFocus] = useState<boolean>(false);

  return (
    <focusTool.Provider value={{ focus, setFocus }}>
      {children}
    </focusTool.Provider>
  );
};
