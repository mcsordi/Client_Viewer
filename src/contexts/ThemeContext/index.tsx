import React, { useCallback, useState } from 'react';
import { themeContext } from './context.ts';
import { Children } from '../../shared/types/Children.ts';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
