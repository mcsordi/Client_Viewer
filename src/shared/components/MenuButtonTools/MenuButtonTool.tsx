import React, { useContext, useReducer } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';
import { focusTool } from '../../../contexts/FocusTool/context.ts';
import { reducer } from './useReducer/reducer.ts';
import { values } from './useReducer/state.ts';

type TButtonToll = {
  text?: string;
  icon?: React.ReactNode;
  type?: string;
  loading?: boolean;
};

export const MenuButtonTool: React.FC<TButtonToll> = ({
  text,
  icon,
  type,
  loading,
}) => {
  const { light } = useContext(themeContext);
  const { focus, setFocus } = useContext(focusTool);

  const [state, dispatch]: any = useReducer<any>(reducer, values);

  return (
    <button
      className={`${light && `bg-slate-200 text-black`} ${!light && !loading && `bg-zinc-800 text-white`}
    flex items-center px-3 py-2 gap-2 uppercase rounded-md font-roboto
    ${!focus && !loading && `first:bg-amber-400 first:text-white first:hover:text-white`} font-medium hover:border
    ${!loading && `hover:border-amber-400 focus:bg-amber-400 hover:text-amber-400 focus:text-white`}
    ${loading && `animate-pulse text-opacity-0 text-slate-200 bg-slate-200`}  loading hover:stroke- border transition-all`}
      onFocus={() => setFocus(true)}
      onBlur={() => setFocus(false)}
      onClick={() => dispatch({ type: type })}
    >
      {icon}
      <span className="hidden sm:flex">{text}</span>
    </button>
  );
};
