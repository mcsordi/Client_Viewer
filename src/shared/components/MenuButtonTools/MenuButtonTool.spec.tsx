import { render, screen } from '@testing-library/react';
import { MenuButtonTool } from './MenuButtonTool';
import { PiPlusBold } from 'react-icons/pi';
import { themeContext } from '../../../contexts/ThemeContext/context';
import { focusTool } from '../../../contexts/FocusTool/context';

describe('<MenuButtonTools/>', () => {
  it('should render a MenuButtonTool component', () => {
    render(<MenuButtonTool text="star" icon={<PiPlusBold />} />);
    const buttonText = screen.getByText('star');
    expect(buttonText.parentElement).toBeInTheDocument();
  });
  it('should render a MenuButtonTool component in the white theme', () => {
    const changeTheme = () => {};
    render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <MenuButtonTool text="star" icon={<PiPlusBold />} />
      </themeContext.Provider>,
    );
    const buttonText = screen.getByText('star');
    expect(buttonText.parentElement).toHaveClass(`bg-slate-200 text-black`);
  });
  it('should render a MenuButtonTool component on the dark theme', () => {
    const changeTheme = () => {};
    render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <MenuButtonTool text="star" icon={<PiPlusBold />} />
      </themeContext.Provider>,
    );
    const buttonText = screen.getByText('star');
    expect(buttonText.parentElement).toHaveClass(`bg-zinc-800 text-white`);
  });
  it('should render a MenuButtonTool with bg-amber-400', () => {
    const setFocus = () => {};
    render(
      <focusTool.Provider value={{ focus: false, setFocus }}>
        <MenuButtonTool text="star" icon={<PiPlusBold />} />
      </focusTool.Provider>,
    );
    const buttonText = screen.getByText('star');
    expect(buttonText.parentElement).toHaveClass(
      `first:bg-amber-400 first:text-white`,
    );
  });
  it('should match a snapshot', () => {
    const setFocus = () => {};
    render(
      <focusTool.Provider value={{ focus: false, setFocus }}>
        <MenuButtonTool text="star" icon={<PiPlusBold />} />
      </focusTool.Provider>,
    );
    const buttonText = screen.getByText('star');
    expect(buttonText.parentElement).toMatchSnapshot();
  });
});
