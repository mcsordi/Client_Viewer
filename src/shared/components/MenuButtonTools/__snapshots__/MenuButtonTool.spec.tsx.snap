// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuButtonTools/> should match a snapshot 1`] = `
<button
  class="bg-zinc-800 text-white
    flex items-center px-3 py-2 gap-2 uppercase rounded-md font-roboto
    first:bg-amber-400 first:text-white font-medium hover:border
     hover:border-amber-400 focus:bg-amber-400 focus:text-white
    hover:text-amber-400 hover:stroke- border transition-all"
>
  <svg
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 256 256"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M228,128a12,12,0,0,1-12,12H140v76a12,12,0,0,1-24,0V140H40a12,12,0,0,1,0-24h76V40a12,12,0,0,1,24,0v76h76A12,12,0,0,1,228,128Z"
    />
  </svg>
  <span
    class="hidden sm:flex"
  >
    star
  </span>
</button>
`;
